### PR TITLE
feat(nx-monorepo): map workspace scripts to nx run-many

### LINKED ISSUE #247: NxMonorepo scripts/tasks should be for workspace operations
Currently nx-monorepo project uses the default project tasks (build/precompile/compile/watch/etc) which don't have much value for the root workspace. These should be using workspace operations utilizing `npx nx run-many` format.

https://github.com/aws/aws-prototyping-sdk/blob/f4f5401f37aa0a81918114bf4c62560c4636e5b0/package.json#L3-L20

Expected:
- build: `npx nx run-many --target=build --all --parallel=4 --output-style stream --nx-bail`
- test: `npx nx run-many --target=test --all --parallel=4 --output-style stream --nx-bail`
- ...

Once this is completed, the github workflows could use the rather than having inline for consistency and maintainability.

### PR DESCRIPTION
**Pull request by JeremyJonas**
Fixes #247

### CODE REVIEW

File: packages/nx-monorepo/src/nx-monorepo.ts [hunk 1]
```diff
@@ -156,9 +197,34 @@ export class NxMonorepoProject extends TypeScriptProject {
...
-    this.compileTask.reset();
-    this.testTask.reset();
+    // Add alias task for "projen" to synthesize workspace
+    this.addTask("synth-workspace", {
+      exec: "npx projen",
+      description: "Synthesize workspace",
+    });
+
+    this.addTask("run-many", {
+      receiveArgs: true,
```
Comment (agdimech) [inline packages/nx-monorepo/src/nx-monorepo.ts:207]: oo00oo is this property new?

File: packages/nx-monorepo/src/nx-monorepo.ts [hunk 2]
```diff
@@ -156,9 +197,34 @@ export class NxMonorepoProject extends TypeScriptProject {
...
+    });
+
+    // Map tasks to nx run-many
+    this._overrideNxBuildTask(this.buildTask, { target: "build" }, true);
+    this._overrideNxBuildTask(this.compileTask, { target: "compile" });
+    this._overrideNxBuildTask(this.preCompileTask, { target: "pre-compile" });
+    this._overrideNxBuildTask(this.postCompileTask, { target: "post-compile" });
+    this._overrideNxBuildTask(this.eslint?.eslintTask, { target: "eslint" });
+    this._overrideNxBuildTask(this.packageTask, { target: "package" });
+    this._overrideNxBuildTask(this.packageTask, { target: "package" });
```
Comment (agdimech) [inline packages/nx-monorepo/src/nx-monorepo.ts]: duplicate.

File: packages/nx-monorepo/src/nx-monorepo.ts [hunk 3]
```diff
@@ -239,6 +305,92 @@ export class NxMonorepoProject extends TypeScriptProject {
...
+   * Helper to format `npx nx run-many ...` style command.
+   * @param options
+   */
+  public formatNxRunManyCommand(options: NxRunManyOptions): string {
+    const cmd: string[] = [
+      "npx nx run-many",
+      `--target=${options.target}`,
+      `--output-style=${options.outputStyle || "stream"}`,
+    ];
+    if (options.configuration) {
```
Comment (agdimech) [inline packages/nx-monorepo/src/nx-monorepo.ts:342]: These options can be simplified to `options.configuration && cmd.push(...)`

File: packages/nx-monorepo/src/nx-monorepo.ts [hunk 4]
```diff
@@ -239,6 +305,92 @@ export class NxMonorepoProject extends TypeScriptProject {
...
+
+    task.description += " for all affected projects";
+
+    return task;
+  }
+
+  /**
+   * Add project task that executes `npx nx run-many ...` style command.
+   */
+  public addNxRunManyTask(name: string, options: NxRunManyOptions): Task {
```
Comment (agdimech) [inline packages/nx-monorepo/src/nx-monorepo.ts:411]: I assume this is so clients can add additional tasks wrapped in nx?

File: packages/nx-monorepo/src/nx-monorepo.ts [hunk 5]
```diff
@@ -239,6 +305,92 @@ export class NxMonorepoProject extends TypeScriptProject {
...
+    if (task == null) {
+      return;
+    }
+
+    if (force) {
+      // @ts-ignore - private property
+      task._locked = false;
+    }
+
+    task.reset(this.formatNxRunManyCommand(options), {
```
Comment (agdimech) [inline packages/nx-monorepo/src/nx-monorepo.ts:399]: Are we going to run into issues with clobbering if a user changed any one of these tasks?

File: packages/nx-monorepo/src/nx-monorepo.ts (hunk 1 above)
Comment (JeremyJonas) [reply packages/nx-monorepo/src/nx-monorepo.ts:207]: Yes it came out in Nov last year - great additional - https://projen.io/api/API.html#addtaskname-props-

File: packages/nx-monorepo/src/nx-monorepo.ts (hunk 2 above)
Comment (JeremyJonas) [reply packages/nx-monorepo/src/nx-monorepo.ts]: Fixed

File: packages/nx-monorepo/src/nx-monorepo.ts (hunk 3 above)
Comment (JeremyJonas) [reply packages/nx-monorepo/src/nx-monorepo.ts:342]: True - but the lines will break regardless and likely be less readable

File: packages/nx-monorepo/src/nx-monorepo.ts (hunk 4 above)
Comment (JeremyJonas) [reply packages/nx-monorepo/src/nx-monorepo.ts:411]: Yes that is correct

File: packages/nx-monorepo/src/nx-monorepo.ts (hunk 5 above)
Comment (JeremyJonas) [reply packages/nx-monorepo/src/nx-monorepo.ts:399]: Can you elaborate on potential issues you are thinking of?

**Review by agdimech — DISMISSED** [review]

**Review by agdimech — APPROVED** [review]

**Pull request merged**